src/Signal.lua

```diff
@@ -47,15 +48,46 @@ function Signal.new()
 end
 
 function Signal:connect(callback)
+	if typeof(callback) ~= "function" then
+		error("Expected the listener to be a function.")
+	end
+
+	-- TODO: subscribe can't happen while store isDispatching, but can't get that store state here
+	if self._store and self._store.isDispatching then
+		error(
+			'You may not call store.changed:connect() while the reducer is executing. ' ..
+			  'If you would like to be notified after the store has been updated, subscribe from a ' ..
+			  'component and invoke store:getState() in the callback to access the latest state. '
+		)
+	end
+
 	local listener = {
 		callback = callback,
 		disconnected = false,
+		connectTraceback = debug.traceback(),
+		disconnectTraceback = nil
 	}
 
 	self._listeners = immutableAppend(self._listeners, listener)
 
 	local function disconnect()
+		if listener.disconnected then
+			local errorMessage = ("Listener connected at: \n%s\n" ..
+				"was already disconnected at: \n%s\n"):format(
+					tostring(listener.connectTraceback),
+					tostring(listener.disconnectTraceback)
+			)
+			self._store.errorReporter:reportErrorDeferred(errorMessage, debug.traceback())
```

Comment:
A lot of stacktraces involved here 😅

Reply:
if there's a double unsubscribe, knowing where the first unsubscribe happened -- in addition to the second one -- will be necessary for figuring out if there's some component lifecycle (or other) issue.